app/models/application.rb

```diff
@@ -0,0 +1,41 @@
+class Application < ApplicationRecord
+  validates :name, presence: true
+  validates :street_address, presence: true
+  validates :city, presence: true
+  validates :state, presence: true
+  validates :zipcode, presence: true, numericality: true
+  validates :description, presence: true
+  has_many :application_pets
+
+
+  def self.application(id)
+    Application.find(id)
+  end
+
+  def full_address
+    street_address << " " << city << ", " << state << " " << zipcode
+  end
+
+  def list_of_pets
+    Pet.joins(:applications).where("application_id = ?", self.id)
```

Comment:
This query is equivalent to `self.pets` -- it's a bit easier/more readable just to use the associations rather than use a join